sysconfig/sysconfig.go

```diff
@@ -53,15 +54,44 @@ type Options struct {
 	GadgetSnap snap.Container
 }
 
-// FilesystemOnlyApplyOptions is the set of options for
-// ApplyFilesystemOnlyDefaults.
-type FilesystemOnlyApplyOptions struct {
-	// Classic is true when the system in rootdir is a classic system
-	Classic bool
+// Device carries information about the device model and mode that is
+// relevant to sysconfig.
+type Device interface {
+	RunMode() bool
+	Classic() bool
+
+	Kernel() string
+	//Base() string
```

Comment:
A TODO maybe depending on whether we actually need Base and Model?

Reply:
I'll take care of this in a follow up one way or another. I want to cover at least one of the TODO I left in configcore. The one where we still read the kernel command line.